temporalio/activity.py

```diff
@@ -0,0 +1,287 @@
+"""Functions that can be called inside of activities.
+
+Most of these functions use :py:mod:`contextvars` to obtain the current activity
+in context. This is already set before the start of the activity. Activities
+that make calls that do not automatically propagate the context, such as calls
+in another thread, should not use the calls herein unless the context is
+explicitly propagated.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import contextvars
+import logging
+import threading
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    Mapping,
+    MutableMapping,
+    NoReturn,
+    Optional,
+    Tuple,
+)
+
+import temporalio.exceptions
+
+
+@dataclass(frozen=True)
+class Info:
+    """Information about the running activity.
+
+    Retrieved inside an activity via :py:func:`info`.
+    """
+
+    activity_id: str
+    activity_type: str
+    attempt: int
+    heartbeat_details: Iterable[Any]
+    heartbeat_timeout: Optional[timedelta]
+    schedule_to_close_timeout: Optional[timedelta]
+    scheduled_time: datetime
+    start_to_close_timeout: Optional[timedelta]
+    started_time: datetime
+    task_queue: str
+    task_token: bytes
+    workflow_id: str
+    workflow_namespace: str
+    workflow_run_id: str
+    workflow_type: str
+    # TODO(cretz): Add headers, current_attempt_scheduled_time, retry_policy, and is_local
```

Comment:
There are so many things to add, it would be 1000 issues. The difficulty with ever reaching PR status is this requirement that you have to implement everything or create an issue for it before PR.